operations/tempo-mixin/dashboards/rollout-progress.libsonnet

```diff
@@ -0,0 +1,318 @@
+      // Because of this we can't use the mixin's addRow() and addPanel().
+      schemaVersion: 27,
+      rows: null,
+      panels: [
+        //
+        // Rollout progress
+        //
+        $.panel('Rollout progress') +
+        $.barGauge([
+          // Multi-zone deployments are grouped together removing the "zone-X" suffix.
+          // After the grouping, the resulting label is called "cortex_service".
+          |||
+            (
+              sum by(cortex_service) (
+                label_replace(
+                  kube_statefulset_status_replicas_updated{%(namespace_matcher)s,statefulset=~"%(all_services_regex)s"},
+                  "cortex_service", "$1", "statefulset", "(.*?)(?:-zone-[a-z])?"
+                )
+              )
+              /
+              sum by(cortex_service) (
+                label_replace(
+                  kube_statefulset_replicas{%(namespace_matcher)s},
+                  "cortex_service", "$1", "statefulset", "(.*?)(?:-zone-[a-z])?"
+                )
+              )
+            ) and (
+              sum by(cortex_service) (
+                label_replace(
+                  kube_statefulset_replicas{%(namespace_matcher)s},
+                  "cortex_service", "$1", "statefulset", "(.*?)(?:-zone-[a-z])?"
+                )
+              )
+              > 0
+            )
+          ||| % config,
+          |||
+            (
+              sum by(cortex_service) (
+                label_replace(
+                  kube_deployment_status_replicas_updated{%(namespace_matcher)s,deployment=~"%(all_services_regex)s"},
+                  "cortex_service", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"
+                )
+              )
+              /
+              sum by(cortex_service) (
+                label_replace(
+                  kube_deployment_spec_replicas{%(namespace_matcher)s},
+                  "cortex_service", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"
+                )
+              )
+            ) and (
+              sum by(cortex_service) (
+                label_replace(
+                  kube_deployment_spec_replicas{%(namespace_matcher)s},
+                  "cortex_service", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"
+                )
+              )
+              > 0
+            )
+          ||| % config,
+        ], legends=[
+          '{{cortex_service}}',
+          '{{cortex_service}}',
+        ], thresholds=[
+          { color: 'yellow', value: null },
+          { color: 'yellow', value: 0.999 },
+          { color: 'green', value: 1 },
+        ], unit='percentunit', min=0, max=1) + {
+          id: 1,
+          gridPos: { h: 8, w: 10, x: 0, y: 0 },
+        },
+
+        //
+        // Writes
+        //
+        $.panel('Writes - 2xx') +
+        $.newStatPanel(|||
+          sum(rate(cortex_request_duration_seconds_count{%(gateway_job_matcher)s, route=~"%(gateway_write_routes_regex)s",status_code=~"2.+"}[$__rate_interval])) /
```

Comment:
I think all these `cortex_` metrics should be `tempo_` metrics? Double check by importing the dashboard into an env that has tempo to make sure these graphs are populated.